src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String inputFile = readFile(fromFileName);
+        writeToFile(inputFile, toFileName);
+    }
+
+    private String readFile(String fileName) {
+        File file = new File(fileName);
+        int supplyNum = 0;
+        int buyNum = 0;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String value = bufferedReader.readLine();
+
+            while (value != null) {
+                stringBuilder.append(value).append(" ");
+                value = bufferedReader.readLine();
+            }
+
+            String[] dataArray = stringBuilder.toString().split(" ");
+
+            for (String string : dataArray) {
+                String[] arrayForNum = string.split(",");
+                if (arrayForNum[0].equals(SUPPLY)) {
+                    supplyNum += Integer.parseInt(arrayForNum[1]);
+                } else {
+                    buyNum += Integer.parseInt(arrayForNum[1]);
+                }
+            }
+
+            return createReport(supplyNum, buyNum);
+
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file" + file, e);
+        }
+    }
+
+    private String createReport(int supply, int buy) {
+        int result;
+        StringBuilder answer = new StringBuilder();
+        String separator = System.lineSeparator();
+        result = supply - buy;
+        answer.append(SUPPLY).append(COMMA).append(supply).append(separator);
+        answer.append(BUY).append(COMMA).append(buy).append(separator);
+        answer.append(RESULT).append(COMMA).append(result);
+
+        return answer.toString();
+    }
+
+    private void writeToFile(String data, String resultFile) {
+        File file = new File(resultFile);
+        file.delete();
```

Comment:
The same. Do you really need to delete file explicitly ? Please investigate it

Reply:
not fixed